src/source/Ice/SocketConnection.c

```diff
@@ -281,6 +289,68 @@ STATUS socketConnectionReadData(PSocketConnection pSocketConnection, PBYTE pBuf,
     return retStatus;
 }
 
+STATUS socketConnectionClosed(PSocketConnection pSocketConnection)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE;
+
+    CHK(pSocketConnection != NULL, STATUS_NULL_ARG);
+
+    MUTEX_LOCK(pSocketConnection->lock);
+    locked = TRUE;
+
+    pSocketConnection->connectionClosed = TRUE;
```

Comment:
Can this simply be an atomic? Or is the idea that potentially we can add more stuff here?

Reply:
changed to atomic